src/gen/conditions.rs

```diff
@@ -767,6 +766,65 @@ fn cond_test(input: &str) -> Result<(Allocator, Vec<SpendConditionSummary>), Val
     cond_test_cb(input, None)
 }
 
+#[test]
+fn test_invalid_condition_list1() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 (8 )))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_condition_list2() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 ((8 ))))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_condition_list_terminator() {
+    let (a, spend_list) = cond_test("((({h1} ({h2} (123 (((80 (50 8 ))))").unwrap();
+
+    assert_eq!(spend_list.len(), 1);
+    assert_eq!(*spend_list[0].coin_id, test_coin_id(VEC1, VEC2, 123));
+    assert_eq!(a.atom(spend_list[0].puzzle_hash), VEC2);
+
+    assert_eq!(spend_list[0].seconds_relative, 50);
+}
+
+#[test]
+fn test_invalid_condition_short_list_terminator() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 (((80 8 ))))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_spend_list1() {
+    assert_eq!(
+        cond_test("(8 )").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_spend_list2() {
+    assert_eq!(
+        cond_test("((8 ))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_spend_list_terminator() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 (()) 8 ))").unwrap_err().1,
```

Comment:
Is this `(h1 h2 123 (()) 8)` ?

I understand this notation can have one or two values within a paren pair, and if the second one is omitted, it is filled in a s nil.

The sequence `(123 (()) 8 )` seems to have 3 top-level elements.

`cond_test("((({h1} ({h2} (123 (() 8 )))").unwrap_err().1,` ?